### ISSUE TITLE
IndexOutOfBoundsException in WindowSketchQuantiles

### ISSUE THREAD

**Issue by trustin** [micrometer-metrics/micrometer#139]
Still using 0.10.0, but I believe the issue is still there because `WindowSketchQuantiles` was not updated since long ago:

```
java.lang.IndexOutOfBoundsException: Index: 2330176, Size: 2330176
 at java.util.LinkedList.checkElementIndex(LinkedList.java:555) at java.util.LinkedList.get(LinkedList.java:476)
 at io.micrometer.core.instrument.stats.quantile.WindowSketchQuantiles.get(WindowSketchQuantiles.java:103)
 at io.micrometer.core.instrument.prometheus.internal.CustomPrometheusSummary$Child.collect(CustomPrometheusSummary.java:109)
 at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:267)
 at java.util.concurrent.ConcurrentLinkedQueue$CLQSpliterator.forEachRemaining(ConcurrentLinkedQueue.java:851)
 at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
 at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
 at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
 at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
 at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
 at io.micrometer.core.instrument.prometheus.internal.CustomPrometheusSummary.collect(CustomPrometheusSummary.java:152)
 at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.findNextElement(CollectorRegistry.java:180)
 at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.nextElement(CollectorRegistry.java:213)
 at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.nextElement(CollectorRegistry.java:134)
 at io.prometheus.client.exporter.common.TextFormat.write004(TextFormat.java:22)
 ...
```

**Comment by trustin** [comment]
FYI, We monitor the following: 0.5, 0.75, 0.95, 0.98, 0.99, 0.999, 1.0

**Comment by trustin** [comment]
On the other hand, it seems to me that using a `LinkedList` doesn't seem like a good idea.

**Comment by trustin** [comment]
We started to observe performance issue with `WindowSketchQuantiles`. The server is stuck at the following stack trace when Prometheus attempts to fetch the metrics from the exposition service:

```
"armeria-common-worker-epoll-3-1" #29 daemon prio=5 os_prio=0 tid=0x00007f8786960000 nid=0x90e2 runnable [0x00007f87ecf4f000]
   java.lang.Thread.State: RUNNABLE
    at java.util.ComparableTimSort.mergeLo(ComparableTimSort.java:732)
    at java.util.ComparableTimSort.mergeAt(ComparableTimSort.java:481)
    at java.util.ComparableTimSort.mergeCollapse(ComparableTimSort.java:406)
    at java.util.ComparableTimSort.sort(ComparableTimSort.java:213)
    at java.util.Arrays.sort(Arrays.java:1246)
    at java.util.Arrays.sort(Arrays.java:1433)
    at java.util.List.sort(List.java:478)
    at java.util.Collections.sort(Collections.java:141)
    at io.micrometer.core.instrument.stats.quantile.WindowSketchQuantiles.getFinalSummary(WindowSketchQuantiles.java:297)
    at io.micrometer.core.instrument.stats.quantile.WindowSketchQuantiles.get(WindowSketchQuantiles.java:100)
    at io.micrometer.core.instrument.prometheus.internal.CustomPrometheusSummary$Child.collect(CustomPrometheusSummary.java:109)
    at io.micrometer.core.instrument.prometheus.internal.CustomPrometheusSummary$$Lambda$644/521088971.apply(Unknown Source)
    at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:267)
    at java.util.concurrent.ConcurrentLinkedQueue$CLQSpliterator.forEachRemaining(ConcurrentLinkedQueue.java:851)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
    at io.micrometer.core.instrument.prometheus.internal.CustomPrometheusSummary.collect(CustomPrometheusSummary.java:152)
    at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.findNextElement(CollectorRegistry.java:180)
    at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.nextElement(CollectorRegistry.java:213)
    at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.nextElement(CollectorRegistry.java:134)
    at io.prometheus.client.exporter.common.TextFormat.write004(TextFormat.java:22)
    at ...
```

**Comment by jkschneider** [comment]
Fixed with 736fd95ca5a9fb9846699489877eb4021882b9df
